packages/next-safe-action/src/hook.ts

```diff
@@ -163,27 +166,23 @@ export const useOptimisticAction = <const Schema extends z.ZodTypeAny, const Dat
 			setOptimisticState(input);
 			setInput(input);
 
-			return executor
+			executor
 				.current(input)
-				.then((res) => setResult(res))
+				.then((res) => (res ? setResult(res) : void 0))
```

Comment:
Wouldn't it be better to use `DEFAULT_RESULT` here instead of `void 0` if `res` is undefined?